### PR TITLE
core: turn kernel_init.c and panic.c into submodules of core

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR changes the default kernel_init to be a submodule called "core_init".
That makes it possible to completely bypass it, as done by riotboot.
Previously, riotboot depended on a specific link order to properly override ```kernel_init()```
```core_init``` is made a default module, riotboot simply disables it.

Same with ```panic.c``` turned into the ```core_panic``` module.

### Testing procedure

- basic compilation and execution should still work
- riotboot should be tested
- actual code sizes should be unchanged

CI can do that.

### Issues/PRs references

#13238

### CODE REVIEW

File: bootloaders/riotboot/Makefile [hunk 1]
```diff
@@ -14,6 +14,7 @@ CFLAGS += -DRIOTBOOT
 
 # Disable unused modules
 CFLAGS += -DNDEBUG -DLOG_LEVEL=LOG_NONE
+DISABLE_MODULE += core_init core_msg
```
Comment (nmeum) [inline bootloaders/riotboot/Makefile]: What about `core_panic`, shouldn't it be disabled as well?

Comment (kaspar030) [reply bootloaders/riotboot/Makefile]: > What about `core_panic`, shouldn't it be disabled as well?

Yes, that'll be another PR.

Comment (kaspar030) [reply bootloaders/riotboot/Makefile]: > Yes, that'll be another PR.

actually, that doesn't make sense. They both require exactly the same testing.

**Comment by kaspar030** [comment]
- rebased, changed panic.c similar to kernel_init.c

@nmeum Does this fix the issues with linktime re-ordering?

**Comment by nmeum** [comment]
> @nmeum Does this fix the issues with linktime re-ordering?

Yes, it does seem to fix it, did a few tests locally. I also just pushed your two commits on the #13176 link time reordering branch. Since I can just rebase my branch against master as soon as this PR is merged. If you want, you can wait to see if it passes on the CI.

**Comment by kaspar030** [comment]
Thanks @nmeum @haukepetersen!

**Pull request merged**